src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -87,7 +87,7 @@ def validate_environment(self, attrs, source):
 
     def save(self, rule):
         rule.project = self.context['project']
-        if self.data.get('environment'):
+        if 'environment' in self.data:
```

Comment:
Does this return an environment id? I'm not sure I can see how the updated test accounts for the prior falsey check.

Reply:
Yes, `self.data['environment']` is an environment ID (or `None`), if that's what you are asking.

The issue was that it wasn't possible to pass a `null`/`None` environment to this serializer and have it actually make any changes, since this block would never be entered. The test previously didn't actually test that this field changed (it went from `None` to `None`) so the updated test ensures a `None` value can override a non-`None` value.

Reply:
awesome, thanks for the explanation 🥇